src/main/java/com/juniorstart/juniorstart/repository/UserProfileRepository.java

```diff
@@ -6,12 +6,14 @@
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
-
 import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
 
 @Repository
 public interface UserProfileRepository extends JpaRepository<UserProfile, Long> {
 
+    Optional<UserProfile> findByPrivateId(UUID id);
```

Comment:
To chyba jest domyślnie. w JpaRepository masz findById i też zwraca optionala.

Reply:
Tak, ale domyślnie tylko po Longu ID